### PR TITLE
configurable line ending normalization

### PR DESCRIPTION
**Pull request by pfitzseb**
Fixes https://github.com/domluna/JuliaFormatter.jl/issues/98.

### CODE REVIEW

**Comment by domluna** [comment]
LGTM only question I have is if the option is in a config file do you need to do any extra for will the symbol be read correctly 

As a reference https://github.com/domluna/JuliaFormatter.jl/pull/365/files#diff-8712d022b5e49f56d5b2d2b3ff1d7f072ae2e818741ef43d4eacc9084c4c2a3fL566-L579 here the style string is converted to the appropriate object.

**Comment by pfitzseb** [comment]
Yeah, dunno. Switched to strings instead, which should definitely work :)

File: src/JuliaFormatter.jl [hunk 1]
```diff
@@ -358,7 +370,15 @@ function format_text(cst::CSTParser.EXPR, style::AbstractStyle, s::State)
...
 
     text = String(take!(io))
-    text = normalize_line_ending(text)
+
+    replacer = if s.opts.normalize_line_endings === "unix"
+        WINDOWS_TO_UNIX
+    elseif s.opts.normalize_line_endings === "windows"
+        UNIX_TO_WINDOWS
+    else
+        choose_line_ending_replacer(s.doc.text)
```
Comment (pfitzseb) [inline src/JuliaFormatter.jl:379]: This is the original text, right? Is there a better way to get that?

Comment (domluna) [reply src/JuliaFormatter.jl:379]: not atm I think it's only referred to in one other place.

I think using the text arg in the function should work too - you would just have to not override it like we do right now on line 372

Comment (pfitzseb) [reply src/JuliaFormatter.jl:379]: This function only takes `cst::CSTParser.EXPR, style::AbstractStyle, s::State`, so we don't have that in here :)

Comment (domluna) [reply src/JuliaFormatter.jl:379]: oops misread that, then yeah s.doc.text is the original text :)

**Review by domluna — APPROVED** [review]
this seems g2g, anything else to do?

**Comment by pfitzseb** [comment]
No this is good from my end.

**Pull request merged**